File: algorithms/[1603]设计停车系统/solution.ts
// topics = ["设计"]

class ParkingSystem {
  park: [number, number, number];
  constructor(big: number, medium: number, small: number) {
    // 剩余车位
    this.park = [big, medium, small];
  }

  /**
   * time O(1), space O(1)
   */
  addCar(carType: number): boolean {
    if (this.park[carType - 1] == 0) {
      return false;
    }
    this.park[carType - 1] -= 1;
    return true;
  }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * var obj = new ParkingSystem(big, medium, small)
 * var param_1 = obj.addCar(carType)
 */

File: algorithms/[1603]设计停车系统/solution.py
# topics = ["设计"]


class ParkingSystem:
    """
    time O(1), space O(1)
    """
    def __init__(self, big: int, medium: int, small: int):
        self.park = [big, medium, small]

    def addCar(self, carType: int) -> bool:
        if self.park[carType - 1] == 0:
            return False
        self.park[carType - 1] -= 1
        return True


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)
